### DISCUSSION TITLE
Seek to a specific time in media file

### DISCUSSION THREAD

**Discussion by therentabrain** [WarmUpTill/SceneSwitcher#995] [Q&A] [answered]
Happy user and recommender of Scene Switcher for >2 years now.  I have around 20 macros and I use most of them every stream!

I have a 12 hour long video file of a clock.  0:00:00 in the file is noon/midnight, 6:00:00 shows 6 o'clock, and so on.

The video plays continuously during streams and keeps the time perfectly.  I have to set it manually to the current time before I start every stream, unless I leave OBS running (without crashes) 24/7. :)

How to achieve this on OBS startup without having to do it by hand?

I have a medium-okay method but it isn't great.  I wrote a formula that figures out the value of the current time in hours plus two decimal places, and I use Media Seek To Duration.  But I have to copy and paste the value in.  

The field for Duration won't allow a variable or formula.  Otherwise, I'd make a macro that runs a VBS script to get the hour decimal calculation and save it in a text file, and have Switcher read that text value into a variable which would be used for Duration.

That's kludgey though.  It would be awesome to just tell it 'seek to the current time' or 'seek to the hours, minutes, and seconds in this variable' or something more direct.  But I'll take any solution, as long as it works.  It's frustrating to have to set the time again whenever OBS crashes and I'm trying to get my stream back on.

Thanks!

**Comment by UUoocl** [comment, marked as answer]
Here's an idea. In the macro Conditions include a Date condition that is ignored. 
In the Actions, create a Variable using the 'Set to condition value' option and choose the Date condition.  This will store the current date and time in a variable.
The time can be extracted from the variable using substring or Math expression.  
The variable can then be used in the Media action.      

<img width="761" alt="image" src="https://github.com/WarmUpTill/SceneSwitcher/assets/99063397/c566444f-20d9-42f0-b421-5290fb6d708e">

**Reply by therentabrain** [reply]
That looks perfect!  Is that a beta or something?  I don't have the ability to put a variable in Seek to duration, and I thought I had the latest Switcher!

**Reply by therentabrain** [reply]
I also don't recognize the style/shape of the double chevron buttons, but could just be the skin

**Reply by therentabrain** [reply]
![image](https://github.com/WarmUpTill/SceneSwitcher/assets/6657421/696e5d4b-63db-46c4-9862-77c760dfde47)

**Reply by therentabrain** [reply]
I figured it out.  Once I have a variable defined, it becomes possible to change the box by clicking the vertical ellipsis.  Trying to make it work now

**Reply by therentabrain** [reply]
Okay, that was ... complicated.  But it works great.
Here's what I did:

![image](https://github.com/WarmUpTill/SceneSwitcher/assets/6657421/dae802b9-a67d-4bbf-8256-9fa16cedb307)
![image](https://github.com/WarmUpTill/SceneSwitcher/assets/6657421/c7987043-62ff-4ec4-a768-6c7807d91ed2)
![image](https://github.com/WarmUpTill/SceneSwitcher/assets/6657421/32eaabb5-66dd-47a4-ad0d-282a79ac60d6)
![image](https://github.com/WarmUpTill/SceneSwitcher/assets/6657421/469f3cba-29e5-4068-b6fd-dcbca0a27aeb)

I did not see a simpler way to do it.  But it works great!  Thank you!

**Reply by UUoocl** [reply]
Happy to help! thanks for sharing the solution.

**Reply by therentabrain** [reply]
Followup:
This would only work if I had a 24 hour long file, but since it's an analog clock, it's just 12 hours.
This will account for the difference in the formula:

`((${Hours}*60*60)+(${Minutes}*60)+${Seconds}) % 43200`

The % stands for mod/modulo/modulus, and will keep the answer within bounds, because there are 43200 seconds in 12 hours.

**Reply by therentabrain** [reply]
Update:  This almost works great, but the digit position changes when the date goes from 1 digit to 2 digits, or vice versa.  So just now when it had recently changed from the 31st to the 1st, I had to change all the positions by 1 character, and in 4 days I'll have to do it again, and (the key part) not forget, because there's no error message, it just fails silently, and the clock starts at 12am.  

Worse yet, if I stream past midnight on the night of the 9th, it will work prior but not if I have to redo it / restart OBS / reload my scene collection after.

I solved it by writing a VBScript that does the math and places the result in a text file.

VBScript:

```
Option Explicit

  On Error Resume Next
    Dim T
    T = (Timer + 4) Mod 43200

    With CreateObject("Scripting.FileSystemObject").CreateTextFile("c:\temp\calctime.t", True, True)
      .Write(T)
      .Close
    End With
  On Error Goto 0
```

![image](https://github.com/WarmUpTill/SceneSwitcher/assets/6657421/710895b3-3a30-4769-95da-7ae20480f0cc)

![image](https://github.com/WarmUpTill/SceneSwitcher/assets/6657421/6c4ead36-f0dc-4af5-8453-8fc6df1a82f3)

Notes:
Waiting 1 second is useful because it takes around 500ms for the Condition to detect that the txt file has changed.  
Using Write is necessary because when I used WriteLine, the newline in the file caused the clock set to fail.  This could be something Switcher could be smarter about.  But it's ok, it works now.
I have it adding 4 seconds to Timer to compensate for delays and stream lag, but it may be just fine to take that away.

**Reply by WarmUpTill** [reply]
Sorry I somehow missed this discussion thread until now.
Just to add for completeness (your current method is completely fine):

You could also to set up a helper macro which defines various date related variables, which you then use to perform your math operation on.

The macro should look something like this:
![image](https://github.com/WarmUpTill/SceneSwitcher/assets/19472752/96876180-bc5e-47df-87c2-9eb8494b02ba)

You can [import](https://github.com/WarmUpTill/SceneSwitcher/wiki/Exporting-and-importing-individual-macros) this example using this code:

```
AAAVXnic7VhbbyI3FP4rlp/abaAMm220vG2zRIu0aVDIpqqqqDIzhxmLwR7ZHhYa8d97jj03GnVLSx4HgTT+zsXn8tmD/cw3Ijba8snvz1yJDfAJX4BjH4UD9iiMFMscLL/ghSgtClcit0AjI/Ic8gbQ6joTKm017FoW0x3Ed2rhhHENnhpdFs0o0fEap3NSpRjCMzeQSuvANAqZsPel+rl0Tis+caYM2JyieYHiTK60n8WyE1gm0yzHn5utrrVKpJNa2Qe0aDRM7f8Bdhgnx/nqfI9wPydKSvVS9iVgKI2PJgkRVUrt/EwYYBT28MjihgL6lomPeMgPF6FrH2ISNTqLIpcOizfX1o99TwvqbfT+3eGier56d3V4qhxM0V1w8s+2V9H7xnZElnWPPmm3hr3tcAIrEEC0fmoK1YWcTtMc5n+H2xL4eWOd56KwkDS+JT7yBDmJ9cp1KmOMBdlTGkFWtzqRK0mseeZObuAP9GedEVI5r2eBJvAE24qcWt88jIYod/uCHg8UsgwmWzDWlyE6IJyI/d3qV4C1lzXRhiAwqgcZFk6p2A0s2SUbjSbR28koYuPR+JK3SuNvaslUaQO09NrEPRT8B54bKEC066ksyPeduq/goFRX5n+m3Mn4OoN4XbstBHEEVfgQP+z4y4kcIv7XNm6rXQUTBkUPSe2+lvwS9qHfQBjeouMKnilMRCbM4/kSmIUc/LSojG1/DGlyHKly89hJutO46CeiRboB5WbJjk8GEY7L5cIZ3IqqHWvUheSf4BFkP+zmbR3e8Aq7FS7OvC/UWkmVoCEq0EJmTjNCvGqRixiOZRXIvkqX1e5wVa9kSm0LRepuvQ7zrmumi6rgI78ruGy6KwzY0En+HYvYD2zM3rC37Hv2IyOW4dq7Lq3Tm7nRm6JlklRF6WqMf0AXqWKeMxRj3YVJ8DDzyhR1pvOks2HLFwI+GGCVwTCH6Q4Gvu1babSi4j8et/zLYno/v7+7mX2eUjNjULBoukurO9D2on5V0VboYFOgm1YcEdFCHzzh9sSjA21i5zLyVivfoZ6SPSXPouTGE+lVOPlR7HtG9ow8l5H4wn8dPn7SZf/W7gl5NiEz4tHrvLWlKv3Joedkz8nzXtuBSa/CyoU/lPas7Fl5LivD9QayEs/f0Nzo0Bn8iaA6s6M7vupsbcW2Ug93GbASZe4ahhHP1fHR53SL8Mf0dP3qj8N/CKne1083aRbd6SYPYN0JBk+dCxQeDceXwzE//AVvTSre
```

**Reply by therentabrain** [reply]
Aww, well that'd work.  Maybe I'll switch to that method sometime, thank you!

**Reply by therentabrain** [reply]
Ok I tried it.  It looks like it will work fine for setting the clock, but is there a way to get two digit values with a leading 0? 

Maybe there's a better way to accomplish this.  Should I start a new topic?  My task:

I use the Source Record plugin to record two cameras (not at the same time, but on different segments of a stream).  I want the Source Record filenames to have matching date and timestamps to the overall stream recording's filename.  I imagine I can use SS to set that based on a variable that remembers when I pressed Record.  But, I need the files to be in YYYY MM DD HH-MM-SS format, because I have other things that read the filenames by column.

Thanks for any advice :)
